addons/mail/static/tests/discuss/call/call.test.js

```diff
@@ -746,3 +746,47 @@ test("single 'join' (with camera) button when last call had camera on", async ()
         contains: [".fa-video-camera"],
     });
 });
+
+test("dynamic focus switches to talking participant", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "General" });
+    const aliceSessionId = pyEnv["discuss.channel.rtc.session"].create({
+        channel_member_id: pyEnv["discuss.channel.member"].create({
+            channel_id: channelId,
+            partner_id: pyEnv["res.partner"].create({ name: "Alice" }),
+        }),
+        channel_id: channelId,
+    });
+    const bobSessionId = pyEnv["discuss.channel.rtc.session"].create({
+        channel_member_id: pyEnv["discuss.channel.member"].create({
+            channel_id: channelId,
+            partner_id: pyEnv["res.partner"].create({ name: "Bob" }),
+        }),
+        channel_id: channelId,
+    });
+
+    const env = await start();
+    const rtc = env.services["discuss.rtc"];
+    await openDiscuss(channelId);
+    await click("[title='Join Call']");
+    await click(".o-discuss-CallParticipantCard[title='Alice']");
+    await contains(".o-discuss-Call-mainCards .o-discuss-CallParticipantCard[title='Alice']");
+    await contains(".o-discuss-Call-mainCards .o-discuss-CallParticipantCard[title='Bob']", {
+        count: 0,
+    });
+
+    // Check if focus switches to Bob (the one that is talking)
+    rtc.updateSessionInfo({ [bobSessionId]: { isTalking: true } });
+    await contains(".o-discuss-Call-mainCards .o-discuss-CallParticipantCard[title='Bob']");
+
+    // Check if we switch to the most recent speaker
+    rtc.updateSessionInfo({ [aliceSessionId]: { isTalking: true } });
+    await contains(".o-discuss-Call-mainCards .o-discuss-CallParticipantCard[title='Bob']", {
+        count: 0,
+    });
+    await contains(".o-discuss-Call-mainCards .o-discuss-CallParticipantCard[title='Alice']");
+
+    // check if we switch back to the last recent speaker
+    rtc.updateSessionInfo({ [aliceSessionId]: { isTalking: false } });
+    await contains(".o-discuss-Call-mainCards .o-discuss-CallParticipantCard[title='Bob']");
```

Comment:
Would remove comments and empty newlines: the test is self-explanatory and it's best to keep LOCs cohesive per test. This makes it easier to read code inside test suite (test scopes are better defined, and each test is efficient in LOCs so that less scrolling is necessary to read it all)